stylefunction.js

```diff
@@ -66,7 +67,9 @@ export function getValue(layer, layoutOrPaint, property, zoom, feature) {
     let value = (layer[layoutOrPaint] || emptyObj)[property];
     const propertySpec = spec[`${layoutOrPaint}_${layer.type}`][property];
     if (value === undefined) {
-      value = propertySpec.default;
+      value = (propertySpec.default !== undefined) ? propertySpec.default : 'none';
+    } else if ((typeof value == 'object') && !('default' in value)) {
+      value.default = (propertySpec.default !== undefined) ? propertySpec.default : '';
```

Comment:
While this fixes the `inline-geojson` example, it might break in cases where the expected value is not a string. A better default here would be `null`, which is handled by the expression engine, but `icon-image` cannot be null, There is an open issue for this: mapbox/mapbox-gl-js#6935.